packages/library-base/src/commonMain/kotlin/io/realm/kotlin/schema/RealmPropertyType.kt

```diff
@@ -63,3 +63,9 @@ public data class ListPropertyType(
     override val storageType: RealmStorageType,
     override val isNullable: Boolean = false
 ) : RealmPropertyType
+
+// List and Set could be the same type but made them different for visibility
```

Comment:
Maybe worth briefly considering it already because some thoughts/constraints around dictionaries could reveal nice concepts for the set type that could be hard to change without breaking it ... but I guess we already have the issue with the list type.